### ISSUE TITLE
Find people you may know

### ISSUE THREAD

**Issue by rubensworks** [SolidLabResearch/Challenges#9]
## Pitch

Based on the people listed in my profile, I want to find other people I _may_ know, based on the people my contacts know.

## Desired solution

- An algorithm for detecting "people you may know", ordered by relevance
  - The algorithm should recursively find people other people know by following links between documents, where each recursion should produce "people you may know" with a lower priority.
  - If multiple people know the same person, the chance that you know that person is increased.
  - The algorithm should be "streaming" and "lazy", so that new people can be requested only when they are required.
- A Web application that lists people you may know based on the new algorithm.
  - It should show a list of people that can be infinitely scrolled, where new people are only requested from the algorithm if they need to be displayed.
- Different ontologies for encoding "people you know" should be supported.

## Acceptance criteria

- An application in which you can provide your WebID and obtain an infinitely scrolling list of people you may know.
- Basic details (name +WebID) of people are shown
- This demo should work for people with different (RDF) data storage formats (all friends in profile doc, friends in separate doc linked from profile, ...) and ontologies (Schema.org, foaf, ...)

## Pointers

- Link traversal: https://www.rubensworks.net/blog/2022/01/21/querying-a-decentralized-web/

**Comment by bjdmeest** [comment]
considering the last acceptance criteria, doesn't this depend on #15 then?

**Comment by rubensworks** [comment]
Yes, that specific criterion would depend on #15 indeed.
